### ISSUE TITLE
DP #38 Fair and Secure Private Real Estate sales

### ISSUE THREAD

**Issue by EndymionJkb** [district0x/district-proposals#38]
**Name**: Fair and Secure Private Real Estate Sales

**Purpose**: Reduce cost and reliance on lawyers by automating many processes involved in a private sale of real estate, including escrow of funds. (This could apply to real estate in general, but the legal and business interests regulating public sales (i.e., through Realtors) is so entrenched and powerful, I believe it would have to start with private sales between individuals.)

**Description**: In most jurisdictions, private sellers are required to comply with the same disclosure and other laws as a real estate broker listing the property. A system of smart contracts could -- provably and immutably -- ensure compliance with these laws and even store the documents (e.g., using IPFS or Storj or similar service). It doesn't need to be all that fast: one characteristic of real estate transactions is they happen slowly, over days and weeks (even the closing takes hours). It's not like a POS system that has to process transactions in seconds.

The next part of the process is the bidding and negotiation. Again, sellers could choose the type of sale they wanted to have (auction, sealed bid, open bid, what policy for multiple offers, etc.), and contracts could accept bids (including escrowed hand money), and the seller could take action, for instance, accepting a certain bid, which would automatically refund the others and escrow the deposit funds.

Private sellers are still subject to fair housing regulations, just as brokers are, and a system like this, immutably recording offers, could eliminate disputes about who offered what and when, and why someone was denied. The contract rules treat every buyer the same, and don't know about age, ethnicity, gender, etc. It would be hard to argue that a contract was racist!

HUD and auction sites already do something very like this, albeit not on the blockchain, and not with real money. Using Oracles or Slock.it services, it would even be possible to vet buyers in some ways (e.g., checking credit).

The rest of the process could also be mediated through contracts, right up to (and possibly eventually including) the closing itself. For instance, the buyer could call a contract method to specify a home inspector and a radon inspector, who would have known public addresses (ENS would be helpful here). The contract would then accept reports from those addresses. The time periods could all be encoded into the contract so that, for instance, the buyer could call "terminateContract()," which would succeed inside the inspection time period (and release the escrow back to the buyer). After the inspection period, the call would fail.

The lender could also have a public address, and could send a transaction from that known address clearing the loan approval conditions. At closing, the title company could send a transaction saying all the documents were signed and the money disbursed, which would release the escrowed hand money to the title company. All these legal entities would still do their thing (conventional paper processes with legal backing), but at the end, they'd send the contract notification that all the "human/legal stuff" was done, and the transaction would move to a new state.

This would eliminate disputes about timing (I once almost lost $5,000 because I hand delivered a contract termination which they claimed they never received!), and the transaction data could always contain a hash of the actual document (and maybe a key to retrieve it from IPFS or Storj), so there could be no lost documents or disputes about what was signed when by whom. This could cut down on litigation considerably.

Note: I'm a Realtor with 15 years experience, very familiar with the industry and its challenges. I'm also thinking about a system based on the Quorum project (or maybe Hyperledger) for coordinating the various parties needed for a conventional real estate closing. Automation could reduce costs and errors (replacing reconciliation across multiple independent databases with consensus).

**Comment by tiggercr** [comment]
Yes fully support creating something like this for my country Ireland

**Comment by EndymionJkb** [comment]
Great! I'm working on the main contract for this now. There are a lot of subtleties! (I'll publish it when it's more ready for prime time.) I'm thinking the contract itself should not store significant value; there are too many legalities around deposits, and we wouldn't want to present a target to hackers!

The deposit handling would work more like submitting the escrow party's address with an offer, then having that trusted escrow authority send a transaction verifying that the (probably fiat) funds were received, and maybe an IPFS pointer to the actual contract.

There are unavoidable "points of legal contact," such as transfer taxes to be paid, deeds to be recorded, etc., that would prevent full automation -- at least for the foreseeable future. The whole system would have to be on-chain for that to happen; in the meantime, there would need to be something like addresses of trusted authorities who sign off on whatever they're responsible for.

Same issue with something like a blockchain title search. Title searches are done by hand because 1) it might require reading the whole deed and applying human judgment; and 2) somebody needs to take legal liability and write a title insurance policy. You could put all the property records on chain, with IPFS pointers to scans of the documents, and meta-data to automate preliminary searches. But I can't imagine title companies today writing policies that put them on the hook for tens of thousands of dollars without somebody personally verifying the results.

For the sales contract itself, there could be private sales terms, sent as a string encrypted with the seller's public key (which would have to be decrypted by the seller off-chain). I'm also thinking there should be a deposit to make an offer, most of which would be refunded (perhaps minus a small fee for maintaining the system).

You wouldn't want all the offer amounts to be public, so buyers would submit a hash of the number + passphrase (without the passphrase, numbers would hash to the same thing, which leaks information). The deposit would discourage spurious offers/spammers (even more than the regular gas cost).

To accept an offer, the seller would call acceptOffer with the plain text offer amount and passphrase (provided by the buyer; presumably they've been talking all along). The number and passphrase should then hash to the same value the buyer originally submitted, proving to all that this was the actual offer presented. 

After the property sold, buyers could all withdraw their deposits - with a time limit unlock in case the contract got stuck somehow: e.g., the seller never accepted any offer.

**Comment by AngeloAdam** [comment]
please send me a direct message on district slack regarding your proposal. @angelo

**Comment by rd2madnss** [comment]
I think this idea is great and exactly why ETH was designed as it is, based on my limited understanding of smart contracts thus far. However I think the goal should be more far reaching and intended to be ubiquitous for all property in time. Truly it will be a very lengthy process and probably never ever completed. But as time goes on and the database grows it should become many orders of magnitude more efficient, reliable, and accurate than our current systems for clearing titles and checking for liens on property before transfer. Foremost I think the goal should be to represent large titles first: real estate, vehicles, aircraft, vessels, etc. in a public ledger as smart contracts that also represent value. Having one decentralized system should be preferable to many, IMHO. Perhaps this DP, DP #144, or another marketplace could have it's goals modified and be intended to fill this role. Whatever the case my feeling is that the goal should be to have one standard marketplace, not hundreds of different ones. Competition in this space will drive innovation but we all know how important being the first on the block is. I will make a similar post to this one in Hyperplace and you can bet I will be staking whatever marketplace I can find that has this intention.

Best Regards,

R
